### DISCUSSION TITLE
numba#6148 seems closed, is numpy vector numba implementation still incomplete?

### DISCUSSION THREAD

**Discussion by meliache** [scikit-hep/vector#156] [General]
The vector readthedocs documentation says:
> The Vector library includes extensions to inform Numba about vector objects, vector NumPy arrays, and vector Awkward Arrays. At the time of writing, the implementation of vector NumPy arrays is incomplete due to [numba/numba#6148](https://github.com/numba/numba/pull/6148).

I just checked the status of that PR and it has been closed and supplanted by [numba#7121](https://github.com/numba/numba/pull/7121) which has been already merged. Is this statement still up-to-date? Or was that numba#7121 not enough? Is there any work required on vector side?

**Comment by jpivarski** [comment]
It sounds like we no longer have a blocker then. The one feature that was missing from Vector is to pass NumPy arrays of vectors into and out of Numba—you can already do it with Awkward Arrays.

The NumPy-array-of-vectors case would add functionality: NumPy arrays are mutable. But other than that, viewing the NumPy arrays as Awkward Arrays would function as a work-around now.
